src/OrchardCore.Modules/OrchardCore.Users/Localization/Providers/UserLocalizationRequestCultureProvider.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Localization;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Localization;
+
+namespace OrchardCore.Users.Localization.Providers;
+
+public class UserLocalizationRequestCultureProvider : RequestCultureProvider
+{
+    public override async Task<ProviderCultureResult> DetermineProviderCultureResult(HttpContext httpContext)
+    {
+        ArgumentNullException.ThrowIfNull(httpContext);
+
+        if (httpContext?.User?.Identity?.IsAuthenticated == false)
+        {
+            return null;
```

Comment:
```suggestion
            return NullProviderCultureResult;
```

Reply:
I agree about returning `NullProviderCultureResult`

Reply:
This won't compile

Reply:
I answered too quickly, I thought it was an aspnet thing

aspnet providers return null

Reply:
Need to leave will complete the review toonight ;)

Reply:
just add `await` before `NullProviderCultureResult` :)

Reply:
> I answered too quickly, I thought it was an aspnet thing

It's a `protected` field, check this PR #13182